docs/proposals/inclusion-of-notes.md

```diff
@@ -0,0 +1,92 @@
+# Inclusion of notes Proposal <!-- omit in TOC -->
+
+Currently it is not possible within Foam to include other notes into a note. Next to including a full note it could be interesting to add functionalities that allow for greater flexibility. This proposal discusses some functionalities around inclusion of notes.
+
+**IMPORTANT: This design is merely a proposal of a design that could be implemented. It DOES NOT represent a commitment by `Foam` developers to implement the features outlined in this document. This document is merely a mechanism to facilitate discussion of a possible future direction for `Foam`.**
+
+- [Introduction](#introduction)
+- [New features](#new-features)
+  - [Including a note](#including-a-note)
+  - [Include a section of a note](#include-a-section-of-a-note)
+  - [Include an attribute of a file (note property or frontmatter)](#include-an-attribute-of-a-file-note-property-or-frontmatter)
+  - [Inclusion of backlinks overview](#inclusion-of-backlinks-overview)
+  - [Introducing a query language](#introducing-a-query-language)
+
+## Introduction
+
+Initial work and thought on including a note was ignited by issue [#652](https://github.com/foambubble/foam/issues/652). Requested by a user was a likewise functionality as offered in Obsidian. This was simply the ability to include a note.
+
+Whilst researching digital gardening for my own setup, I came across an in-depth overview by [Maggie Appleton](https://maggieappleton.com/roam-garden). Showing examples of her personal Roam Research I see valuable possibilites to connect more information, if we would add additional functionalities to the possibility of including a note. This proposal displays these possible functionalities and markup.
+
+## New features
+
+### Including a note
+
+The minimal functionality is the ability to fully include a note. Markup used in Obsidian for this is `![[wikilink]]`. For Foam I would suggest to follow this syntax. Benefits being:
+
+- Adds minimal amount of knowledge required as syntax is based on the syntax of creating a wikilink.
+- Makes the auto-complete work ouf-of-the-box, without any additional code and listeners required.
+
+**Important**. A risk exists that a loop of including the same notes arises. E.g. Note A includes note B which includes note A. This needs to be prevented by the implementation and made visible to the user.
+
+### Include a section of a note
+
+It could be interesting to only include a section of a note instead of the entire note. In order to do so thse user should be able to use the following syntax:
+
+`![[wikilink#section-b]]`
```

Comment:
So from observed usage in Dendron, I would say that this would be the desirable default behavior. We introduced the syntax of header ranges and wildcard block headers to facilitate this effect: https://wiki.dendron.so/notes/f1af56bb-db27-47ae-8406-61a98de6c78c.html#wildcard-headers-as-a-range-boundary

Something else you might want to consider is removing the header in a reference. It's usually not needed.  We also have a concept of an `offset` which moves the reference by `offset` lines down. This all comes together in the following link, which roughly reads - include `section-b`, skip the first line, go to next block

```
![[wikilink#section-b,1:#*]]
```

This is the most commonly used sort of reference in Dendron.  Moving forward, we'll probably set an option to make this default behavior:
```
 `![[wikilink#section-b]]` is alias for  `![[wikilink#section-b,1:#*]]`
```

Reply:
Good point! Can imagine, the section header preferably is left out. I would propose to take that as a default and enable offsetting at a later stage. 

@riccardoferretti what do you think?

Reply:
yup I agree!